source/MaterialXFormat/Util.cpp

```diff
@@ -234,6 +234,86 @@ void flattenFilenames(DocumentPtr doc, const FileSearchPath& searchPath, StringR
     }
 }
 
+FileSearchPath getResolvedDefinitionPath(const FileSearchPath& userDefinitionPath, bool includeSubFolders)
+{
+    FileSearchPath resolvedDefinitionPath;
+    FileSearchPath assetDefinitionPath = getAssetDefinitionPath();
+    for (FileSearchPath::ConstIterator path=assetDefinitionPath.begin(); path != assetDefinitionPath.end(); ++path)
+    {
+        resolvedDefinitionPath.append(*path);
+    }
+    for (FileSearchPath::ConstIterator path=userDefinitionPath.begin(); path != userDefinitionPath.end(); ++path)
+    {
+        resolvedDefinitionPath.append(*path);
+    }
+    if (includeSubFolders)
+    {
+        std::set<std::string> childDefinitionPathsSet;
+        FileSearchPath childDefinitionPaths;
+        for (FileSearchPath::ConstIterator path = resolvedDefinitionPath.begin(); path != resolvedDefinitionPath.end(); ++path)
+        {
+            FilePath filePath = *path;
+            FilePathVec subdirs = filePath.getSubDirectories();
+            for (FilePath subdirPath : subdirs)
+            {
+                if (childDefinitionPathsSet.count(subdirPath.asString()) == 0)
+                {
+                    childDefinitionPaths.append(subdirPath);
+                    childDefinitionPathsSet.emplace(subdirPath.asString());
+                }
+            }
+        }
+        return childDefinitionPaths;
+    }
+    else
+    {
+        return resolvedDefinitionPath;
+    }
+}
+
+FileSearchPath getResolvedTexturePath(const FileSearchPath& userTexturePath, const FileSearchPath& userDefinitionPath, bool includeSubFolders)
+{
+    FileSearchPath assetTexturePath = getAssetDefinitionPath();
```

Comment:
I'm not sure why definition path logic is redone here. It should call the getResolveDefintiionPath() and texture paths should come before definition paths.
Better to double check: https://wiki.autodesk.com/pages/viewpage.action?spaceKey=meecs&title=MaterialX+Pathing+Requirements